# Assignment 3

### Instructions
1. Import the following modules:
    - `pandas`
    - `stats` from `scipy`
2. Create a dataframe from the data contained in *income_data.csv* using `pandas`

In [ ]:
# Code here
import pandas as pd
from scipy import stats
df = pd.read_csv("./income_data.csv")


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *education*

In [ ]:

r, p = stats.pearsonr(df['income'], df['education'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:  {p:.3f}')
print('')


r (Pearson correlation coefficient): 0.725
p-value:  0.000



3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *prestige*

In [ ]:
r, p = stats.pearsonr(df['income'], df['prestige'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:  {p:.3f}')
print('')

r (Pearson correlation coefficient): 0.838
p-value:  0.000



4. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *education* and *prestige*

In [ ]:
# Code here
r, p = stats.pearsonr(df['education'], df['prestige'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:  {p:.3f}')
print('')

r (Pearson correlation coefficient): 0.852
p-value:  0.000



5. In the empty `Markdown` cell below, explain the following for **each of the calculated correlation coefficients**
    - The **strength** and **direction** of the correlation
    - The **statistical significance** of the test (**p-value**); emphasize if we can trust the correlation test or not, and why

The correlation coeffciant for income and education indicate that the two have a strong relationship. A value of 0.725 means that when one goes up so does the other. The P value of 0.000 means it is very unlikely that these results happened by luck, the test is trustworthy. 

The correlation coeffciant for income and prestige indicates the two have an even stronger relationship. A value of 0.838 indicates just like above, the two move in the same direction. The P-Value for this relationship is also 0.000, meaning it is very trustworthy. A low P Value means it is most likely a valid result.

The correlation coeffciant for education and prestige indicate this has the strongest relationship of all. The value is 0.852, meaning they are very closely related. This value is also positive, so the two go in the same direction. The p-value is also 0.000, so it is trust worthy as well.

In the empty `Markdown` cell below, answer the following question:

6. Pearson's correlation coefficient OR covariance can both be used to measure the direction of the relationship between two variables (e.g., education and income). Explain why Pearson's correlation coefficient may be better than covariance.

Pearson's correlation coefficient is better because it measures the strength of the relationship, rather than covariance, which just measures how the variables change together. Pearson's is also more normalized, the value can only be between -1 to +1. In Covariance, it can be from negative infinity to positive infinity.

7. Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment3.ipynb** 
    - Replace '**Last**' and '**First**' with your name